### PR TITLE
adbfs-rootless: 2016-10-02 -> unstable-2023-03-21

### LINKED ISSUE #230091: Update request: adbfs-rootless 2016-10-02 → 2023-03-21
- Package name: adbfs-rootless
- Latest released version: 2023-03-21

- Current version on the unstable channel: 2016-10-02
- Current version on the stable/release channel: 2016-10-02

- [X] Checked the [nixpkgs pull requests](https://github.com/NixOS/nixpkgs/pulls)

**Notify maintainers**

@Profpatsch 

-----

Note for maintainers: Please tag this issue in your PR.

### PR DESCRIPTION
**Pull request by Aleksanaa**
###### Description of changes

Resolves https://github.com/NixOS/nixpkgs/issues/230091

Tested, fully works

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/mobile/adbfs-rootless/default.nix [hunk 1]
```diff
@@ -1,31 +1,30 @@
-{ lib, stdenv, fetchFromGitHub, fetchpatch, pkg-config, fuse, adb }:
+{ lib
+, stdenv
+, fetchFromGitHub
+, pkg-config
+, fuse
+, android-tools
+}:
 
 stdenv.mkDerivation rec {
```
Comment (wegank) [inline pkgs/development/mobile/adbfs-rootless/default.nix:9]:
```suggestion
stdenv.mkDerivation {
```

Comment (Aleksanaa) [reply pkgs/development/mobile/adbfs-rootless/default.nix:9]: But there's still this
```
inherit (src.meta) homepage;
```

File: pkgs/development/mobile/adbfs-rootless/default.nix [hunk 2]
```diff
@@ -1,31 +1,30 @@
...
 
   postPatch = ''
     # very ugly way of replacing the adb calls
-    sed -e 's|"adb |"${adb}/bin/adb |g' \
-        -i adbfs.cpp
+    substituteInPlace adbfs.cpp \
+      --replace '"adb' '"${android-tools}/bin/adb'
   '';
 
   installPhase = ''
```
Comment (wegank) [inline pkgs/development/mobile/adbfs-rootless/default.nix:30]: Probably needs `runHook` here.

Comment (Aleksanaa) [reply pkgs/development/mobile/adbfs-rootless/default.nix:30]: idk, it's `postPatch` not `patchPhase`, why is `runHook` needed?

Comment (wegank) [reply pkgs/development/mobile/adbfs-rootless/default.nix:30]: Not there, but in `installPhase`.

**Comment by wegank** [comment]
There seems to be something like

```
  adbfs-rootless = callPackage ../development/mobile/adbfs-rootless {
    adb = androidenv.androidPkgs_9_0.platform-tools;
  };
```

in `pkgs/top-level/all-packages.nix`, I guess that override should be removed / updated as well?

**Comment by wegank** [comment]
Result of `nixpkgs-review pr 231040` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>adbfs-rootless</li>
  </ul>
</details>

**Review by wegank — APPROVED** [review]

**Pull request merged**